### PR TITLE
[#1326] Move Carousel to components lib

### LINKED ISSUE #1326: Move Carousel to components lib
We have a `Carousel` component that is part of our `render` library. We need to move that to our components library.

### PR DESCRIPTION
**Pull request by bitboxer**
closes #1326

### CODE REVIEW

**Comment by bitboxer** [comment]
This moves the `Carousel` into the `components` library. 

Also, it now figures out that the last component displayed is not fully shown and will only scroll to a position where it is visible. Before today, it didn't care and scrolled right pass it 😆 

https://user-images.githubusercontent.com/56195/112336277-3e7cb580-8cbd-11eb-9045-d2f4bd9e5007.mov

File: frontend/ui/src/pages/Inbox/MessageInput/index.tsx [hunk 1]
```diff
@@ -199,7 +199,7 @@ const MessageInput = (props: MessageInputProps & ConnectedProps<typeof connector
   const getLastMessageWithSuggestedReplies = useCallback(() => {
     const lastMessages = props.messages
       ?.filter((message: Message) => message.senderType == 'source_contact')
-      .slice(props.messages.length - 5)
+      .slice(-5)
```
Comment (bitboxer) [inline frontend/ui/src/pages/Inbox/MessageInput/index.tsx:202]: This was a stupid mistake 🤦

File: frontend/ui/src/pages/Inbox/Messenger/MessengerContainer/index.tsx [hunk 2]
```diff
@@ -32,7 +32,7 @@ const MessengerContainer = ({conversations, currentConversation}: MessengerConta
       ) : (
         <div className={styles.messageDisplay}>
           <MessageList />
-          {currentConversation && <MessageInput Source={currentConversation.channel.source as Source} />}
+          {currentConversation && <MessageInput source={currentConversation.channel.source as Source} />}
```
Comment (bitboxer) [inline frontend/ui/src/pages/Inbox/Messenger/MessengerContainer/index.tsx:35]: I created another ticket for this: #1359

File: frontend/ui/src/pages/Inbox/MessageInput/index.tsx (hunk 1 above)
Comment (chrismatix) [reply frontend/ui/src/pages/Inbox/MessageInput/index.tsx:202]: This is a separate point, but if I re-read the ticket correctly this code also doesn't quite do what is asked.

The difference is subtle, but the original ticket read to me like they want the suggestions to be the union of all the suggestions of the previous 5 messages. Not the first one that has any.

**Review by chrismatix — APPROVED** [review]
lgtm with minor comment

**Pull request merged**